Source/NSMenu.m

```diff
@@ -1626,6 +1626,22 @@ + (void) popUpContextMenu: (NSMenu *)menu
   [menu _rightMouseDisplay: event];
 }
 
+- (void) popUpMenuPositionItem: (NSMenuItem *)item
+                    atLocation: (NSPoint) point
+                        inView: (NSView *) view
+{
+  NSRect cellFrame = [[view window] convertRectToScreen: [view frame]];
```

Comment:
This is only correct if the views frame is in window coordinates. You should first transform to the base view
cellFrame = [view convertRect: [view frame] toView: nil];

Reply:
I believe it always should be.

Reply:
Why? If there are multiple views with different coordinate systems between the window and the NSPathControl this assumption is not true. Believe me, having struggled with all these transformations for years.

Reply:
The change you suggested puts the menu off window

Reply:
What I suggested would look something like this:

 NSRect cellFrame = [view convertRect: [view frame] toView: nil];
 cellFrame = [[view window] convertRectToScreen: cellFrame];

But looking at it again I think that we need to use [view bounds]  instead of [view frame], as the frame is in the parent view coordinate system (Which is why your code works for the simple case). Coordinate transformations are always confusing.

Reply:
Got it and fixed.